coalib/bearlib/abstractions/ExternalBearWrap.py

```diff
@@ -0,0 +1,213 @@
+            """
+            if param_entry[0] != "":
+                return param_entry
+            desc = "No description given."
+            if len(param_entry) == 2:
+                return (desc, param_entry[1])
+            else:
+                desc += " Defaults to " + str(param_entry[2])
+                return (desc, param_entry[1], param_entry[2])
+
+        @classmethod
+        def get_non_optional_params(cls):
+            """
+            Fetches the non_optional_params from ``options['settings']``
+            and also normalizes their descriptions.
+
+            :return:
+                An OrderedDict that is used to create a
+                FunctionMetada object.
+            """
+            non_optional_params = {}
+            for key, value in options['settings'].items():
+                if len(value) == 2:
+                    non_optional_params[key] = cls._normalize_desc(value)
+            return OrderedDict(non_optional_params)
+
+        @classmethod
+        def get_optional_params(cls):
+            """
+            Fetches the optional_params from ``options['settings']``
+            and also normalizes their descriptions.
+
+            :return:
+                An OrderedDict that is used to create a
+                FunctionMetada object.
+            """
+            optional_params = {}
+            for key, value in options['settings'].items():
+                if len(value) == 3:
+                    optional_params[key] = cls._normalize_desc(value)
+            return OrderedDict(optional_params)
+
+        @classmethod
+        def get_metadata(cls):
+            return FunctionMetadata(
+                'run',
+                optional_params=cls.get_optional_params(),
+                non_optional_params=cls.get_non_optional_params())
+
+        @classmethod
+        def _prepare_settings(cls, settings):
+            """
+            Adds the optional settings to the settings dict in-place.
+
+            :param settings:
+                The settings dict.
+            """
+            opt_params = dict(cls.get_optional_params())
+            for key, val in opt_params.items():
+                if key not in settings:
+                    settings[key] = val[2]
+
+        @staticmethod
+        def get_severity(result):
+            """
+            Returns the severity corresponding to the one from the
+            parsed JSON
+
+            :param result:
+                A result dict extracted from the parsed JSON
+
+            :return:
+                A RESULT_SEVERITY
+            """
+            severity_map = {'MAJOR': RESULT_SEVERITY.MAJOR,
+                            'NORMAL': RESULT_SEVERITY.NORMAL,
+                            'INFO': RESULT_SEVERITY.INFO}
+
+            if not 'severity' in result:
+                return RESULT_SEVERITY.NORMAL
+            else:
+                sev = result['severity']
+                return severity_map[sev]
```

Comment:
I wanted that if the user specifies `severity=blablabla` that he would get an Error because that is not a valid severity